cmd/autoscaler/main.go

```diff
@@ -159,12 +164,41 @@ func main() {
 		logger.Fatalw("Failed to start informers", zap.Error(err))
 	}
 
+	// Needs to be called after informers have started.
+	ss, err := statefulsetLister.StatefulSets(ns).Get(component)
+	if err != nil {
+		logger.Fatalw("Failed to get autoscaler StatefulSet", zap.Error(err))
+	}
+
+	// acceptStat is the func to call when this pod owns the given StatMessage.
+	acceptStat := func(sm asmetrics.StatMessage) {
+		collector.Record(sm.Key, time.Now(), sm.Stat)
+		multiScaler.Poke(sm.Key, sm.Stat)
+	}
+	// processStat is the func to process a single StatMessage, accept or forward.
+	processStat := acceptStat
+
+	bucketSize := int(*ss.Spec.Replicas)
+	if bucketSize > 1 {
+		// If there are multiple pods for autoscaler, they must be run in StatefulSet
+		// ordinal assignment leader electior mode. So enable it.
+		ctx = leaderelection.WithDynamicLeaderElectorBuilder(
+			ctx, kubeClient, leaderelection.ComponentConfig{Buckets: uint32(bucketSize)})
+
+		// Build the WebSocket connections for forwarding.
+		wss, withForwarding := forwardingStatsNetworking(bucketSize, acceptStat, logger)
+		processStat = withForwarding
+		// Close the WebSocket connections when shuting down.
+		for _, ws := range wss {
+			defer ws.Shutdown()
+		}
```

Comment:
I'd just return `cancel` from the `forwardingStatsNetworking` that will do that, no need to expose the WS connections outside?

Reply:
done